packages/nep141-erc20/src/natural-nep141/sendToEthereum/index.ts

```diff
@@ -0,0 +1,905 @@
+  mintReceipts: [],
+  lockReceiptBlockHeights: [],
+  lockReceiptIds: [],
+  lockHashes: []
+}
+
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+export const i18n = {
+  en_US: {
+    steps: (transfer: Transfer) => stepsFor(transfer, steps, {
+      [LOCK]: `Start transfer of ${formatLargeNum(transfer.amount, transfer.decimals).toString()} ${transfer.sourceTokenName} from NEAR`,
+      [AWAIT_FINALITY]: 'Confirm in NEAR',
+      [SYNC]: 'Confirm in Ethereum. This can take around 16 hours. Feel free to return to this window later, to complete the final step of the transfer.',
+      [MINT]: `Deposit ${formatLargeNum(transfer.amount, transfer.decimals).toString()} ${transfer.destinationTokenName} in Ethereum`
+    }),
+    statusMessage: (transfer: Transfer) => {
+      if (transfer.status === status.FAILED) return 'Failed'
+      if (transfer.status === status.ACTION_NEEDED) {
+        switch (transfer.completedStep) {
+          case null: return 'Ready to transfer from NEAR'
+          case SYNC: return 'Ready to deposit in Ethereum'
+          default: throw new Error(`Transfer in unexpected state, transfer with ID=${transfer.id} & status=${transfer.status} has completedStep=${transfer.completedStep}`)
+        }
+      }
+      switch (transfer.completedStep) {
+        case null: return 'Transferring from NEAR'
+        case LOCK: return 'Confirming transfer'
+        case AWAIT_FINALITY: return 'Confirming transfer'
+        case SYNC: return 'Depositing in Ethereum'
+        case MINT: return 'Transfer complete'
+        default: throw new Error(`Transfer in unexpected state, transfer with ID=${transfer.id} & status=${transfer.status} has completedStep=${transfer.completedStep}`)
+      }
+    },
+    callToAction: (transfer: Transfer) => {
+      if (transfer.status === status.FAILED) return 'Retry'
+      if (transfer.status !== status.ACTION_NEEDED) return null
+      switch (transfer.completedStep) {
+        case null: return 'Transfer'
+        case SYNC: return 'Deposit'
+        default: throw new Error(`Transfer in unexpected state, transfer with ID=${transfer.id} & status=${transfer.status} has completedStep=${transfer.completedStep}`)
+      }
+    }
+  }
+}
+/* eslint-enable @typescript-eslint/restrict-template-expressions */
+
+/**
+ * Called when status is ACTION_NEEDED or FAILED
+ * @param transfer Transfer object to act on.
+ */
+export async function act (transfer: Transfer): Promise<Transfer> {
+  switch (transfer.completedStep) {
+    case null:
+      try {
+        return await lock(transfer)
+      } catch (error) {
+        console.error(error)
+        if (error.message.includes('Failed to redirect to sign transaction')) {
+          // Increase time to redirect to wallet before recording an error
+          await new Promise(resolve => setTimeout(resolve, 10000))
+        }
+        if (typeof window !== 'undefined') urlParams.clear('locking')
+        throw error
+      }
```

Comment:
why we miss the LOCK case? We do he checkSync in this function, but don't checkFinality

Reply:
`checkFinality` is handled by `checkStatus`. I think `checkSync` was added as a precaution in case the caller tries to act on this transfer but this state should not be reached because the transfer is still 'in-progress' and not 'action-needed' at this step so `checkStatus` is called.